### DISCUSSION TITLE
[FEATURE REQUEST] Restart Windows service that has dependencies

### DISCUSSION THREAD

**Discussion by bsoldhogg** [saltstack/salt#67486] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
When attempting to restart a Windows service that has components that are dependent on **it**, Salt will fail with the error:

`ERROR: Failed To Stop winmgmt: A stop control has been sent to a service that other running services are dependent on.`

**Describe the solution you'd like**
Similar to how Powershell handles this with the force flag, I would like Salt to handle the dependencies so the service can be restarted.

**Describe alternatives you've considered**
At the moment I am using cmd.run with a Powershell command:

`Restart-Service winmgmt -Force`

**Additional context**
WinMgmt isn't the service I'm trying to restart but it has dependencies and suffers the same issue. I've poked around win_service.py and I don't see the stop function handling the dependencies, and part of the error message doesn't live in that code, so it looks like this comes from win32serviceutil but that's as far as I can see.

**Please Note**
If this feature request would be considered a substantial change or addition, this should go through a SEP process here https://github.com/saltstack/salt-enhancement-proposals, instead of a feature request.

**Comment by knine** [comment]
I ran into this today as well for TermService.

**Comment by Danduriel** [comment]
Hi, this feature is also discussed in a bug and has been implemented in the develop branch some time ago. Maybe we can get this going again. 

Discussed in Bug: 
https://github.com/saltstack/salt/issues/56431

Pull Request for the issue: 
https://github.com/saltstack/salt/pull/56574
https://github.com/saltstack/salt/pull/56632

Requested Feature implemented in develop
https://github.com/saltstack/salt/blob/develop/salt/modules/win_service.py#L657
